Sources/swift-parsing-benchmark/HTTP.swift

```diff
@@ -10,75 +10,62 @@ import Parsing
  */
 
 private struct Request: Equatable {
-  let method: String
-  let uri: String
-  let version: String
+  let method: Substring
+  let uri: Substring
+  let version: Substring
 }
 
 private struct Header: Equatable {
-  let name: String
-  let value: [String]
+  let name: Substring
+  let value: [Substring]
 }
 
-private func isToken(_ c: UTF8.CodeUnit) -> Bool {
```

Comment:
Modernizing the style of this benchmark a bit:

- Extending types with helpers instead of free functions
- Inlining `Prefix` work when simple (instead of free functions)
- Embracing `Substring`